apollo-backend/src/main/java/io/logz/apollo/rest/RestServer.java

```diff
@@ -33,16 +34,23 @@
     private final ApolloConfiguration configuration;
     private final Injector injector;
     private final UserDao userDao;
+    private final SlaveService slaveService;
 
     @Inject
-    public RestServer(ApolloConfiguration configuration, Injector injector, UserDao userDao) {
+    public RestServer(ApolloConfiguration configuration, Injector injector, UserDao userDao, SlaveService slaveService) {
         this.configuration = requireNonNull(configuration);
         this.injector = requireNonNull(injector);
         this.userDao = requireNonNull(userDao);
+        this.slaveService = requireNonNull(slaveService);
     }
 
     @PostConstruct
     public void start() {
+        if (slaveService.isRunningInSlaveMode()) {
```

Comment:
Move it to be a configuration flag `disableApiServer`,  should be true only when you start your slave standalone.
We must have api in real deployments